Apromore-Core-Components/Apromore-Portal/src/main/java/org/apromore/portal/controller/PopupLogSubMenuController.java

```diff
@@ -0,0 +1,221 @@
+        item.setLabel("...");
+        item.setStyle(CENTRE_ALIGN);
+        item.setAttribute(SUB_MENU_FOR, subMenuFor);
+        item.addEventListener(ON_CLICK, event -> {
+            try {
+                if (SUB_MENU_FOR_PD.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewPdOrLogFilter(null, false);
+                } else if (SUB_MENU_FOR_LOG_FILTER.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewPdOrLogFilter(null, true);
+                } else if (SUB_MENU_FOR_DASHBOARD.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewAllExistingDashboard();
+                }
+            } catch (Exception ex) {
+                LOGGER.error("Error in forwarding the request", ex);
+            }
+        });
+        menuPopup.appendChild(item);
+    }
+
+    private void viewAllExistingDashboard() {
+        try {
+            Map<String, Object> attrMap = new HashMap<>();
+            attrMap.put("createNewDashboard", false);
+            attrMap.put("forwardFromPopup", true);
+            PortalPlugin plugin = portalPluginMap.get(PluginCatalog.PLUGIN_DASHBOARD);
+            plugin.setSimpleParams(attrMap);
+            plugin.execute(getPortalContext());
+        } catch (Exception e) {
+            LOGGER.error("Error in showing the Dashboard", e);
+        }
+    }
+
+    private void viewExistingDashboard(UserMetadataSummaryType um) {
+        try {
+            Sessions.getCurrent().setAttribute("logSummaries", Collections.singletonList(logSummaryType));
+            Sessions.getCurrent()
+                .setAttribute("userMetadata_dash", userMetaDataUtilService.getUserMetaDataById(um.getId()));
+            Clients.evalJavaScript("window.open('dashboard/index.zul')");
+        } catch (Exception e) {
+            LOGGER.error("Error in showing the Dashboard", e);
+        }
+    }
+
+    private void addMenuItem(Menupopup popup, UserMetadataSummaryType um, boolean visibleOnLoad, String subMenuFor) {
+        Menuitem item = new Menuitem();
+        item.setLabel(um.getName());
+        item.setAttribute(SUB_MENU_FOR, subMenuFor);
+        item.setAttribute(USER_META_DATA, um);
+        item.addEventListener(ON_CLICK, event -> {
+            try {
+                if (SUB_MENU_FOR_PD.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewPdOrLogFilter((UserMetadataSummaryType) event.getTarget().getAttribute(USER_META_DATA), false);
+                } else if (SUB_MENU_FOR_LOG_FILTER.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewPdOrLogFilter((UserMetadataSummaryType) event.getTarget().getAttribute(USER_META_DATA), true);
+                } else if (SUB_MENU_FOR_DASHBOARD.equals(event.getTarget().getAttribute(SUB_MENU_FOR))) {
+                    viewExistingDashboard((UserMetadataSummaryType) event.getTarget().getAttribute(USER_META_DATA));
+                }
```

Comment:
Shouldn't need this conditional code, because the subclass of PopupLogSubMenuController will know whether it's PD, log filter, dashboard, etc.

Reply:
It is not related with sub-class. It is related with event fired from the front-end and then captured from the back-end to distinguish type of event has been fired.

Reply:
Should the subclasses ever receive and have to handle an event for which they are not the SUB_MENU_FOR?  I assumed not, but I may have missed how this works.